### PR TITLE
Ml on gcp time series bqml

### PR DESCRIPTION
**Pull request by munnm**
Hey Chris,
Here is the lab for time series using BQML and AutoML Tables. 
Let me know what you think.

### CODE REVIEW

**Comment by crawles** [comment]
<p>nice work! some comments, mostly minor</p>

File: courses/machine_learning/time_series_prediction/modeling_bqml.ipynb [hunk 1]
```diff
@@ -0,0 +1,1162 @@
+{
```
Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>nit: add period to end of comments</p><p> </p><p>remove reference to MLE</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>why ___ and not actual value?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>could use except(...columns to exclude...) instead. but maybe less readable?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>ORDER BY iteration</p><p> </p><p>below.. I would add in a sub-section. Compute the accuracy of the model on the eval set, comparing to a simple baseline model that just guesses the majority class. E.g., is this model doing better than a naive strategy?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>This will give us a more precise prediction rather than just predicting if the stock will go up, down, or stay the same. Thus we can treat this problem as either a regression problem or a classification problem, depending on the business needs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>ORDER BY iteration</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>why TODO?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

File: courses/machine_learning/time_series_prediction/modeling_keras.ipynb [hunk 2]
```diff
@@ -0,0 +1,201 @@
+{
```
Comment (crawles) [inline courses/machine_learning/time_series_prediction/modeling_keras.ipynb]: <p>define above:</p><p> </p><p><code></code><code></p><p>def _create_split(phase):</p><p>&nbsp;&nbsp;floor, ceiling = 0, 70</p><p>&nbsp;&nbsp;if phase == 'VALID':</p><p>&nbsp;&nbsp;&nbsp;&nbsp;floor, ceiling = 70, 85</p><p>&nbsp;&nbsp;elif phase == 'TEST':</p><p>&nbsp;&nbsp;&nbsp;&nbsp;floor, ceiling = 85, 100</p><p>&nbsp;&nbsp;return '''</p><p>&nbsp;&nbsp;AND MOD(ABS(FARM_FINGERPRINT(symbol)), EVERY_N * 100) &gt;= (EVERY_N * {0})</p><p>&nbsp;&nbsp;AND MOD(ABS(FARM_FINGERPRINT(symbol)), EVERY_N * 100) &lt;&nbsp;(EVERY_N * {1})</p><p>&nbsp;&nbsp;'''.format(floor, ceiling)</p><p></code><code></code></p><p> </p><p>replace this part with</p><p> </p><p><code></code><code></p><p>subsample = _create_split(phase)</p><p></code><code></code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/624/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: courses/machine_learning/time_series_prediction/modeling_bqml.ipynb (hunk 1 above)
Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>done</p>

[the remark this answers is not in the document]
Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>ok. cool.</p>

Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>ah left as a placeholder to fill in later. thanks. </p>

Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>good comment. Added that as well. </p>

Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>Yeah. Good idea. I added something after to compute the accuracy on the eval data using just the majority guess. It's just less than BQML, phew!</p>

Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>thanks. added.</p>

Comment (munnm) [reply courses/machine_learning/time_series_prediction/modeling_bqml.ipynb]: <p>they want us to add #TODO comments mapping each of the learning objectives to the section/code of the notebook that will correspond to the learning objective. This will be the #3 Learning Objective. </p>

File: courses/machine_learning/time_series_prediction/modeling_keras.ipynb (hunk 2 above)
Comment (crawles) [reply courses/machine_learning/time_series_prediction/modeling_keras.ipynb]: <p>wdty?</p>
